src/rust/src/x509/extensions.rs

```diff
@@ -416,6 +416,133 @@ fn encode_scts(ext: &pyo3::Bound<'_, pyo3::PyAny>) -> CryptographyResult<Vec<u8>
     Ok(asn1::write_single(&result.as_slice())?)
 }
 
+fn encode_naming_authority<'a>(
+    py: pyo3::Python<'_>,
+    ka_str: &'a cryptography_keepalive::KeepAlive<pyo3::pybacked::PyBackedStr>,
+    py_naming_authority: &pyo3::Bound<'a, pyo3::PyAny>,
+) -> Result<extensions::NamingAuthority<'a>, CryptographyError> {
+    let py_oid = py_naming_authority.getattr(pyo3::intern!(py, "id"))?;
+    let id = if py_oid.is_truthy()? {
+        Some(py_oid_to_oid(py_oid)?)
+    } else {
+        None
+    };
+    let py_url = py_naming_authority.getattr(pyo3::intern!(py, "url"))?;
+    let url = if py_url.is_truthy()? {
+        let py_url_str = ka_str.add(py_url.extract::<PyBackedStr>()?);
+        asn1::IA5String::new(py_url_str)
```

Comment:
If this returns `None` we want to propagate an exception; that can happen if the user passes a non-ascii string.

Reply:
@alex please check whether the suggested implementation is readable and the error message is descriptive. I added a simple test to guard against non-ascii values. Do I understand it correctly that the same applies to `registration_number` which is a `PrintableString`?

Reply:
Yup!

Reply:
@alex done! Thank you yet again for a helpful review :handshake: